string go_flag
int8 num_transfers
int8 curr_transfers
string done_flag
string test_msg